doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml

```diff
@@ -0,0 +1,119 @@
+@startuml Eclipse iceoryx Component Overview
+skinparam BackgroundColor transparent
+skinparam componentStyle uml2
+
+skinparam component {
+   backgroundColor<<namespace>> yellow
+}
+
+skinparam package {
+   backgroundColor green
+}
+
+skinparam frame {
+   backgroundColor chocolate
+}
+
+' PlantUML package: CMake package
+' PlantUML frame: CMake library
+
+' @todo #590
+' RouDi split, separate files and namespaces for static/dynamic discovery
+' Implement exported libraries (add_library in iceoryx_posh/CMakeLists.txt)
+' RouDiEnvironment shall not be in roudi namespace, but in testing
+
+header Eclipse iceoryx component overview
+
+' POSH: POSIX SHared memory
+' Publish/subscribe IPC communication infrastructure
+package iceoryx_posh {
```

Comment:
What do you think about the following:
- `iceoryx_posh` component handles all shared memory related tasks
- `iceoryx_roudi` is another component which depends of `iceoryx_posh`
- `iceoryx_runtime` is another component as well

Reply:
Yeah, that's a great idea. I'll address this in a follow-up with #539. We need to keep several things in mind. I'll add this to the issue:
|Module| Safety-certified version | Community version |
|---|---|---|
| Service Discovery | Only static with config file | dynamic service discovery |
| Shared Memory | Created statically during init phase | Create objects in shared memory on-the-fly  |

Reply:
I had a chat yesterday with @elBoberido about it. We concluded that the `iceoryx_roudi` for a default pre-built RouDi makes sense. However, `iceoryx_runtime` will only be a library and not a package in order to be backwards compatible with `v1.0.x`. Having thought a bit more about it today, I realized this will clash with #970 :grimacing: 

@elBoberido What do you think?

Reply:
After discussion with @dkroenke we concluded that sticking to one `iceoryx_posh` CMake package will keep the maintenance overhead lower. I updated the diagram accordingly, introducing new libraries inside `iceoryx_posh`.